### PR TITLE
Feat/watching nonexistant dependencies

### PR DESCRIPTION
**Pull request by staeke**
Branched off https://github.com/whitecolor/ts-node-dev/pull/93

Fixes for dependencies on not (yet) existing files

### CODE REVIEW

**Comment by staeke** [comment]
ping @whitecolor - separated this out now

File: lib/index.js [hunk 1]
```diff
@@ -150,7 +153,14 @@ module.exports = function(script, scriptArgs, nodeArgs, opts) {
 
     // Upon errors, display a notification and tell the child to exit.
     ipc.on(child, 'error', function(m) {
-      log.debug('Child error')
+      log.debug('Child error', m)
+      var lastRequire = m.lastRequire
+      if (m.code === 'MODULE_NOT_FOUND' && lastRequire && lastRequire.path.startsWith('.')) {
+        var pathNoExt = path.normalize(path.join(path.dirname(lastRequire.filename), lastRequire.path))
+        var watch = chokidar.watch([pathNoExt + '.ts', pathNoExt + '.tsx'])
```
Comment (jonjaques) [inline lib/index.js]: why not just call `watcher.add()` here? What does chokidar do that `filewatcher()` cant?

Comment (staeke) [reply lib/index.js]: @jonjaques Please enlighten me if I missed something, but `watcher.add()` didn't work with non-existant files when I tried

Comment (jonjaques) [reply lib/index.js]: Naw, was legit curious. 

As I understand it this PR is to handle the case where you're in an existing file and you import something that doesn't yet exist:

import {something} from './nonexistent'

This handles the error that would shut down ts-node-dev and then sets up a watcher for what would be `./nonexistent.ts|tsx`.

I guess what I meant was, both `filewatcher` and `chokidar` are wrappers around the FS.watch, it just seems like conceptually this project should use one or the other. 

So different question— could we switch all the `filewatcher` code to use `chokidar` instead?  Last publish on filewatcher was 3 years ago.

**Comment by whitecolor** [comment]
Why need to add chokidar? And what are "nonexistant dependencies" could you elaborate workflow for this case?

File: lib/index.js (hunk 1 above)
Comment (staeke) [reply lib/index.js]: I'm all for doing that @jonjaques! I think chokidar is more mature

However, I'm no official contributor. Could coordinate if @whitecolor made me

**Comment by staeke** [comment]
@whitecolor I think @jonjaques explained it quite well

1. `A.ts` requires `B.ts` but `B.ts` doesn't exist
2. `B.ts` is created. This should lead to a recompile

That requires "watching" files that don't yet exist. I found no such support outside of chokidar. Again, I'm willing to migrate to using chokidar throughout.

**Comment by staeke** [comment]
ping again @whitecolor

**Comment by whitecolor** [comment]
This doesn't work in the general case, as it is not possible to correctly resolve not existing dependency.

**Pull request closed without merging**